aws-android-sdk-iot/src/main/java/com/amazonaws/mobileconnectors/iot/AWSIotMqttManager.java

```diff
@@ -972,19 +1017,58 @@ public void connect(
             final String customAuthorizer,
             final AWSIotMqttClientStatusCallback statusCallback) {
 
-        if (tokenKeyName == null || token == null || tokenSignature == null) {
+        if (tokenKeyName == null || token == null) {
             throw new IllegalArgumentException("tokenKeyName/token/tokenSignature cannot be null");
         }
 
         this.userStatusCallback = statusCallback;
 
+        final MqttConnectOptions options = new MqttConnectOptions();
+        // Set custom websocket header
+        customWebsocketHeaders.setProperty("X-Amz-CustomAuthorizer-Name", customAuthorizer);
+        customWebsocketHeaders.setProperty(tokenKeyName, token);
+        if (tokenSignature != null) {
+            customWebsocketHeaders.setProperty("X-Amz-CustomAuthorizer-Signature", tokenSignature);
+        }
+        customAuthConnect(options);
+    }
+
+    /**
+     * Initializes the MQTT session and connects to the specified MQTT server
+     * using username and password as credentials.
+     *
+     * @param username Username to authenticate client to the broker.
+     * @param password Password to authenticate client to the broker.
+     * @param statusCallback When a new MQTT session status is received, a function of callback will
+     *                       be called with new connection status.
+     */
+    public void connect(
+            final String username,
+            final String password,
+            final AWSIotMqttClientStatusCallback statusCallback) {
+
+        if (username == null || password == null) {
+            throw new IllegalArgumentException("username and/or password cannot be null.");
+        }
+
+        this.username = username;
+        this.password = password;
```

Comment:
If you pass these into the methods below, do you need to cache these into the class? Or can they just be method-scoped?

Reply:
These need to be set again for reconnection in case initial attempt at connection fails or an existing connection drops due to network issue. At this point it feels like I should just cache options object at class level and method-scope different fields of options object that are used by different flavors of connect method. I can do a separate PR to refactor this manager class since I think that might be little voluminous.